### PR TITLE
Create driver scripts for single command tasks on RSC

### PR DESCRIPTION
**Pull request by dperaza4dustbit**
These are the changes here:

- Remove all logic about infra-only changes. Filtering in npm-registry PLRs based on path.
- Add driver scripts to satisfy RSC requirements to have single step single command requirement

## Summary by Sourcery

Enforce npm OCI workflows to operate only on real package artifacts and add driver scripts plus utilities and tests to support single-command release tasks on RSC.

New Features:
- Introduce driver scripts for npm release orchestration, including extraction and upload workflows suitable for single-command RSC tasks.

Bug Fixes:
- Prevent npm OCI build and promote tasks from producing or pushing empty placeholder artifacts by requiring at least one .tgz package.
- Fail npm promote and related tasks when package status is not has-packages, relying on upstream path filters for infra-only merges.

Enhancements:
- Add helper functions in npm-common.sh to consistently record error status for Tekton result outputs.
- Strengthen npm OCI promote and build tasks with additional validation for presence of package tarballs before pushing artifacts.

Build:
- Extend the utils container build to include a dedicated test stage that runs script unit tests and gates the final image on passing tests.
- Ensure installed utility scripts in the image are executable via a chmod step.

Tests:
- Add a bash-based unit test suite for npm release plumbing scripts, covering error handling, orchestration behavior, and Pulp interaction edge cases.

### CODE REVIEW

File: utils/Containerfile [hunk 1]
```diff
@@ -33,18 +33,32 @@ COPY requirements.txt .
...
+# Gate the final image on the unit-test stage.
+COPY --from=test /tmp/utils-tests-ok /tmp/utils-tests-ok
+
 COPY --from=builder /venv /venv
 COPY --from=cosign /usr/local/bin/cosign /usr/local/bin/cosign
 COPY --from=oras /usr/bin/oras /usr/local/bin/oras
 COPY --from=oras /usr/local/bin/retry /usr/local/bin/retry
 
 COPY scripts/* /usr/local/bin/
+RUN chmod +x /usr/local/bin/*
```
[the remark this answers is not in the document]
Comment (dperaza4dustbit) [reply utils/Containerfile]: Done
